packages/menu/src/useMenu.ts

```diff
@@ -0,0 +1,666 @@
+        if (rtl && isPrevious) {
+          changeType = StateChangeTypes.MenuItemKeyDownPrevious;
+        }
+
+        if (!rtl && isNext) {
+          changeType = StateChangeTypes.MenuItemKeyDownNext;
+        }
+      } else if (key === KEYS.LEFT) {
+        if (rtl && isNext) {
+          changeType = StateChangeTypes.MenuItemKeyDownNext;
+        }
+
+        if (!rtl && isPrevious) {
+          changeType = StateChangeTypes.MenuItemKeyDownPrevious;
+        }
+      } else if (isVerticalArrowKeys || isJumpKey || isAlphanumericChar) {
+        changeType = isAlphanumericChar
+          ? StateChangeTypes.MenuItemKeyDown
+          : StateChangeTypes[toMenuItemKeyDownType(key)];
+        const nextFocusedValue = getNextFocusedValue({
+          value: item.value,
+          key,
+          isAlphanumericChar,
+          isPrevious,
+          isNext
+        });
+
+        payload = {
+          ...(isFocusedValueControlled ? {} : { focusedValue: nextFocusedValue })
+        };
+
+        changes = { focusedValue: nextFocusedValue };
+      }
+
+      if (changeType) {
+        event.preventDefault();
+
+        dispatch({ type: changeType, payload });
+        onChange({ type: changeType, ...changes });
+      }
+    },
+    [
+      rtl,
+      isExpandedControlled,
+      isFocusedValueControlled,
+      isSelectionValueControlled,
+      getNextFocusedValue,
+      getSelectionValue,
+      onChange
+    ]
+  );
+
+  const handleItemMouseEnter = useCallback(
+    value => {
+      const changeType = StateChangeTypes.MenuItemMouseMove;
+
+      dispatch({
+        type: changeType,
+        payload: {
+          ...(isFocusedValueControlled ? {} : { focusedValue: value })
+        }
+      });
+
+      onChange({
+        type: changeType,
+        focusedValue: value
+      });
+    },
+    [isFocusedValueControlled, onChange]
+  );
+
+  /**
+   * Effects
+   */
+
+  /**
+   * Sets a state value to ensure 1 render happens after
+   * opening so a menu item can be focused.
+   */
+  useEffect(() => {
+    setMenuVisible(controlledIsExpanded);
+  }, [controlledIsExpanded, menuRef, environment]);
+
+  /**
+   * Respond to clicks outside the  open menu
+   */
+  useEffect(() => {
+    const win = environment || window;
+
+    if (controlledIsExpanded) {
+      win.document.addEventListener('click', handleMenuBlur, true);
+    } else if (!controlledIsExpanded) {
+      win.document.removeEventListener('click', handleMenuBlur, true);
+    }
+
+    return () => {
+      win.document.removeEventListener('click', handleMenuBlur, true);
+    };
+  }, [controlledIsExpanded, handleMenuBlur, environment]);
+
+  /**
+   * Return focus to trigger when menu closes
+   */
+  useEffect(() => {
+    if (controlledIsExpanded && !returnFocusRef.current) {
+      returnFocusRef.current = triggerRef.current;
+    }
+
+    if (!controlledIsExpanded && returnFocusRef.current) {
+      setTimeout(() => {
+        returnFocusRef.current!.focus();
```

Comment:
I think the focus treatment here is a bit over-aggressive (similar to Garden's current `Menu`) and results in accessibility violations. Refer to the [APG example](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/#ex_label) to see how a user can tab out of an expanded menu to move on to the next element in the tab order vs. getting sucked back into the menu trigger.

Reply:
I recall this being kind of tricky to do but of course, after a few weeks of not looking at it, I fix things to the correct behavior in 5 minutes. 🤦🏻